File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.cpp
class Solution
{
public:
    int maximumLengthSubstring(string s)
    {
        int count[26] = {0};
        int left = 0;
        int maxLen = 0;

        for (int right = 0; right < s.length(); right++)
        {
            char c = s[right];
            int index = c - 'a';
            count[index]++;

            while (count[index] > 2)
            {
                char leftChar = s[left];
                int leftIndex = leftChar - 'a';
                count[leftIndex]--;
                left++;
            }

            maxLen = max(maxLen, right - left + 1);
        }

        return maxLen;
    }
};

File: 3349-maximum-length-substring-with-two-occurrences/maximum-length-substring-with-two-occurrences.py
class Solution:
    def maximumLengthSubstring(self, s: str) -> int:
        count = [0] * 26
        left = 0
        max_len = 0

        for right in range(len(s)):
            c = s[right]
            index = ord(c) - ord('a')
            count[index] += 1

            while count[index] > 2:
                left_char = s[left]
                left_index = ord(left_char) - ord('a')
                count[left_index] -= 1
                left += 1

            max_len = max(max_len, right - left + 1)

        return max_len